### ISSUE TITLE
Error with No such file or directory for test file

### ISSUE THREAD

**Issue by junzou1992** [TestArmada/flank#565]
When I use `xcodebuild build-for-testing` command and create the test zip file, still get error like below when I run `java -jar ./flank.jar firebase test ios run`, the test zip file and .xctestrun file are located in the same directory.

RunTests
  Uploading test.zip ..
Found xctest: ./Build/Products/Debug-iphoneos/UITests-Runner.app/PlugIns/UITests.xctest
nm -U .-/Build/Products/Debug-iphoneos/UITests-Runner.app/PlugIns/UITests.xctest/UITests
Error: 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm: 2/Build/Products/Debug-iphoneos/UITests-Runner.app/PlugIns/UITests.xctest/UITests: No such file or directory.

Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (ftl.cli.firebase.test.ios.IosRunCommand@1c05a54d): java.lang.RuntimeException: Command failed: nm -U /Build/Products/Debug-iphoneos/UITests-Runner.app/PlugIns/UITests.xctest/UITests
	at picocli.CommandLine.execute(CommandLine.java:1168)
	at picocli.CommandLine.access$800(CommandLine.java:141)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1367)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1335)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
	at picocli.CommandLine.run(CommandLine.java:1974)
	at picocli.CommandLine.run(CommandLine.java:1920)
	at ftl.Main$Companion.main(Main.kt:48)
	at ftl.Main.main(Main.kt)
Caused by: java.lang.RuntimeException: Command failed: nm -U /Build/Products/Debug-iphoneos/UITests-Runner.app/PlugIns/UITests.xctest/UITests
	at ftl.util.Bash.execute(Bash.kt:19)
	at ftl.ios.Parse.parseObjcTests$test_runner(Parse.kt:42)
	at ftl.ios.Xctestrun.testsForTarget(Xctestrun.kt:34)
	at ftl.ios.Xctestrun.findTestNames(Xctestrun.kt:94)
	at ftl.ios.Xctestrun.findTestNames(Xctestrun.kt:78)
	at ftl.args.IosArgs$testShardChunks$2.invoke(IosArgs.kt:63)
	at ftl.args.IosArgs$testShardChunks$2.invoke(IosArgs.kt:26)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at ftl.args.IosArgs.getTestShardChunks(IosArgs.kt)
	at ftl.run.IosTestRunner$runTests$2.invokeSuspend(IosTestRunner.kt:38)
	at ftl.run.IosTestRunner$runTests$2.invoke(IosTestRunner.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:186)
	at ftl.run.IosTestRunner.runTests(IosTestRunner.kt:23)
	at ftl.run.TestRunner.runTests(TestRunner.kt:53)
	at ftl.run.TestRunner.newRun(TestRunner.kt:350)
	at ftl.cli.firebase.test.ios.IosRunCommand$run$1.invokeSuspend(IosRunCommand.kt:32)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.kt:116)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:76)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:53)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:35)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at ftl.cli.firebase.test.ios.IosRunCommand.run(IosRunCommand.kt:31)
	at picocli.CommandLine.execute(CommandLine.java:1160)
	... 9 more

This is how my flank.ios.yml file look like
 test: ./Build/Products/test.zip
xctestrun-file: ./Build/Products/*.xctestrun

**Comment by bootstraponline** [comment]
Here's an example I use to test flank:
```bash
#!/bin/bash

set -euxo pipefail

if ! [ -x "$(command -v xcpretty)" ]; then
  gem install xcpretty
fi

DD="dd_tmp"
SCHEME="FunctionalSwiftTests"
ZIP="ui.zip"
PROJ="./Tests/Functional/FunctionalTests.xcodeproj"

rm -rf "$DD"

xcodebuild build-for-testing \
  -allowProvisioningUpdates \
  -project "$PROJ" \
  -scheme "$SCHEME" \
  -derivedDataPath "$DD" \
  -sdk iphoneos \
  | xcpretty

pushd "$DD/Build/Products"
zip -r "$ZIP" *-iphoneos *.xctestrun
popd
mv "$DD/Build/Products/$ZIP" .
```

```bash
#!/usr/bin/env bash
set -ex

cat << 'EOF' > ./flank.ios.yml
gcloud:
  test: ./ui.zip
  xctestrun-file: ./dd_tmp/Build/Products/*.xctestrun
  timeout: 30m
  device:
  - model: iphone8
    version: 12.2

flank:
  max-test-shards: 2

EOF
```

---

Try using a custom derived data folder to see if that helps. It looks like Flank isn't finding the right path to `UITests`. I'm in the #flank channel on the [firebase community slack](https://firebase.community/) if you'd like to pair via screen share.

**Comment by junzou1992** [comment]
@bootstraponline I was using the custom derived data folder when I run `xcodebuild build-for-testing`, and I even deleted the Debug-iphoneos folders because seems Flank was using the UITests from that directory. But still shows this error.

**Comment by bootstraponline** [comment]
I'm happy to fix this once there's a way to reproduce the problem. Probably screen share is easiest since I can walk you through how to breakpoint Flank. This issue seems specific to your app, I'm not sure why.

**Comment by bootstraponline** [comment]
After debugging, it looks like the problem is solved on the latest stable version of Flank (v6.2.2).

**Comment by bootstraponline** [comment]
If you find more issues, let me know and I'll fix them.
